org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/json/adapters/MessageTypeAdapterFactory.java

```diff
@@ -128,11 +128,47 @@ public Message read(JsonReader in) throws IOException {
 			Object params = parseParams(rawParams, method);
 			return createMessage(jsonrpc, id, method, params, result, error);
 		}
+		
+		protected Object parseParams(JsonReader in, String method) throws IOException {
+			JsonToken next = in.peek();
+			if (next == JsonToken.NULL) {
+				in.nextNull();
+				return null;
+			}
+			Type[] parameterTypes = getParameterTypes(method);
+			if (parameterTypes.length == 1) {
+				return fromJson(in, parameterTypes[0]);
+			}
+			if (parameterTypes.length > 1 && next == JsonToken.BEGIN_ARRAY) {
+				List<Object> parameters = new ArrayList<Object>();
+				in.beginArray();
+				int index = 0;
+				while (in.hasNext() && in.peek() != JsonToken.END_ARRAY) {
+					Type parameterType = index < parameterTypes.length ? parameterTypes[index] : null;
+					Object parameter = fromJson(in, parameterType);
+					parameters.add(parameter);
+					index++;
+				}
+				while (index < parameterTypes.length) {
+					parameters.add(null);
+					index++;
+				}
+				if (in.peek() == JsonToken.END_ARRAY) {
```

Comment:
I don't think this check is necessary. Same for `in.peek() != JsonToken.END_ARRAY` in l. 146. The JsonReader should guarantee that the next token is `END_ARRAY` if and only if `in.hasNext()` is false.